ams/optimizer/spark-optimizer/src/main/java/com/netease/arctic/optimizer/spark/SparkOptimizerExecutor.java

```diff
@@ -0,0 +1,122 @@
+
+import java.util.List;
+
+/**
+ * The {@code SparkOptimizerExecutor} takes OptimizingTask from AMS and wraps it as a spark job,
+ * then submit to the spark environment.
+ */
+public class SparkOptimizerExecutor extends OptimizerExecutor {
+  private static final Logger LOG = LoggerFactory.getLogger(SparkOptimizerExecutor.class);
+  private final JavaSparkContext jsc;
+  private final SparkOptimizerConfig sparkOptimizerConfig;
+  private final int threadId;
+
+  public SparkOptimizerExecutor(JavaSparkContext jsc, SparkOptimizerConfig config, int threadId) {
+    super(config, threadId);
+    this.sparkOptimizerConfig = config;
+    this.jsc = jsc;
+    this.threadId = threadId;
+  }
+
+  @Override
+  protected OptimizingTaskResult executeTask(OptimizingTask task) {
+    OptimizingTaskResult result;
+    try {
+      String threadName = Thread.currentThread().getName();
+      long startTime = System.currentTimeMillis();
+      LOG.info("Now [{}] execute task {}", threadName, task);
+      ImmutableList<OptimizingTask> of = ImmutableList.of(task);
+      String jobDesc =
+          String.format("%s | TableName(%s)", task.getTaskId(), parseFullTableName(task));
+      jsc.setJobDescription(jobDesc);
+      SparkOptimizingTaskFunction taskFunction =
+          new SparkOptimizingTaskFunction((SparkOptimizerConfig) getConfig(), threadId);
+      List<OptimizingTaskResult> results = jsc.parallelize(of, 1).map(taskFunction).collect();
+      result = results.get(0);
+      LOG.info(
+          "[{}] execute task {}, completed {} and time {} ms",
+          threadName,
+          task,
+          result,
+          System.currentTimeMillis() - startTime);
+      return result;
+    } catch (Throwable r) {
+      LOG.error("Failed to execute optimizing task.", r);
+      LOG.error("Optimizer executor[{}] executed task[{}] failed", threadId, task.getTaskId(), r);
+      result = new OptimizingTaskResult(task.getTaskId(), threadId);
+      result.setErrorMessage(ExceptionUtil.getErrorMessage(r, 4000));
+      return result;
+    }
+  }
+
+  @Override
+  protected OptimizingTask pollTask() {
+    OptimizingTask task = null;
+    long startTime = System.currentTimeMillis();
+    while (isStarted()) {
+      try {
+        task = callAuthenticatedAms((client, token) -> client.pollTask(token, threadId));
+      } catch (TException exception) {
+        LOG.error("Optimizer executor[{}] polled task failed", threadId, exception);
+      }
+      if (task != null) {
+        LOG.info("Optimizer executor[{}] polled task[{}] from ams", threadId, task.getTaskId());
+        break;
+      } else {
+        waitAShortTime();
+      }
+      if (System.currentTimeMillis() - startTime > sparkOptimizerConfig.getTaskPollingTimeout()) {
+        LOG.warn(
+            "Waited for {} s, the SparkOptimizerExecutor-{} will stop",
+            sparkOptimizerConfig.getTaskPollingTimeout(),
+            threadId);
+        stop();
+      }
+    }
+    return task;
+  }
```

Comment:
"There should be no need to override the methods of the parent class. There is no need to actively exit the Spark driver.